services/src/test/java/edu/unc/lib/dl/cdr/services/rest/modify/ExportCsvIT.java

```diff
@@ -0,0 +1,246 @@
+import edu.unc.lib.dl.fcrepo4.ContentRootObject;
+import edu.unc.lib.dl.fcrepo4.FolderObject;
+import edu.unc.lib.dl.fcrepo4.PIDs;
+import edu.unc.lib.dl.fcrepo4.RepositoryInitializer;
+import edu.unc.lib.dl.fcrepo4.RepositoryObjectFactory;
+import edu.unc.lib.dl.fcrepo4.RepositoryObjectLoader;
+import edu.unc.lib.dl.fcrepo4.RepositoryPIDMinter;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.search.solr.service.ChildrenCountService;
+import edu.unc.lib.dl.search.solr.service.SolrSearchService;
+import edu.unc.lib.dl.test.AclModelBuilder;
+import edu.unc.lib.dl.test.RepositoryObjectTreeIndexer;
+import edu.unc.lib.dl.util.ResourceType;
+
+/**
+ *
+ * @author lfarrell
+ *
+ */
+@ContextHierarchy({
+        @ContextConfiguration("/spring-test/test-fedora-container.xml"),
+        @ContextConfiguration("/spring-test/cdr-client-container.xml"),
+        @ContextConfiguration("/spring-test/solr-indexing-context.xml"),
+        @ContextConfiguration("/export-csv-it-servlet.xml")
+})
+public class ExportCsvIT extends AbstractAPIIT {
+    private static final String COLLECTION_PATH =
+            "/Content Collections Root/Admin unit/Collection";
+    private static final String FOLDER_PATH =
+            "/Content Collections Root/Admin unit/Collection/Folder";
+
+    @Autowired
+    protected String baseAddress;
+    @Autowired
+    protected File solrDataDir;
+    @Autowired
+    protected EmbeddedSolrServer server;
+    @Autowired
+    protected ChildrenCountService childrenCountService;
+    @Autowired
+    protected SolrUpdateDriver driver;
+    @Autowired
+    protected SolrSearchService solrSearchService;
+    @javax.annotation.Resource(name = "accessGroups")
+    protected AccessGroupSet accessGroups;
+    @Autowired
+    protected Model queryModel;
+    @Autowired
+    protected RepositoryObjectLoader repositoryObjectLoader;
+    @Autowired
+    protected RepositoryObjectFactory repositoryObjectFactory;
+    @Autowired
+    protected DocumentIndexingPackageFactory dipFactory;
+    @Autowired
+    protected RepositoryPIDMinter pidMinter;
+    @Autowired
+    private RepositoryInitializer repoInitializer;
+    @Autowired
+    private RepositoryObjectTreeIndexer treeIndexer;
+    @Autowired
+    private RepositoryObjectSolrIndexer solrIndexer;
+
+    protected ContentRootObject rootObj;
+    protected AdminUnit unitObj;
+    protected CollectionObject collObj;
+    protected FolderObject folderObj;
+
+    @Before
+    public void setup() throws Exception {
+        setupContentRoot();
+        generateBaseStructure();
+
+        treeIndexer.indexAll(baseAddress);
+
+        setField(solrSearchService, "solrClient", server);
+        setField(childrenCountService, "solrClient", server);
+
+        solrIndexer.index(rootObj.getPid(),
+                unitObj.getPid(),
+                collObj.getPid(),
+                folderObj.getPid());
+    }
+
+    @Test
+    public void exportCsv() throws Exception {
```

Comment:
There are a few more cases we should test:
* Export container which contains a work with a file (missing file metadata fields was what demonstrated the need for a test in the first place)
* Export from the root
* Export with only one result
* Export containing a described resource
* Export containing a delete resource (I'd suggest deleting a container that contains a child, to verify both the "Deleted" and "Parent Deleted" cases)
* Export a file resource directly (this is disallowed in the admin ui, I'm not really sure what the outcome would be, but returning just the file would be fine, or throwing an error, whichever seems better)

Could you also verify the filename for the csv file itself is set in at least one test?